fix: unicode code points

novusnota opened the pull request:
Closes #531

<!--
IMPORTANT:
If your PR doesn't close a particular issue, please, create the issue first and describe the whole context: what you're adding/changing and why you're doing so. And only then open the Pull Request, which would close that issue!

In case you are adding a new language feature, a standard library function or introducing other user-facing changes, you need to document them via a new PR to tact-docs.
-->

- [x] I have updated CHANGELOG.md
- ~~[ ] I have documented my contribution in Tact Docs: https://github.com/tact-lang/tact-docs/pull/PR-NUMBER~~ Not applicable
- [x] I have added tests to demonstrate the contribution is correctly implemented: this usually includes both positive and negative tests, showing the happy path(s) and featuring intentionally broken cases
- [x] I have run all the tests locally and no test failure was reported
- [x] I have run the linter, formatter and spellchecker
- [x] I did not do unrelated and/or undiscussed refactorings

novusnota committed:
fix: regex

novusnota committed:
feat: better error messages for unicode code points outside of valid range

novusnota committed:
feat: escape unicode control codes in comment strings

novusnota committed:
chore: CHANGELOG

anton-trunov reviewed (changes requested):
We need a test demonstrating the FunC compiler can process comments with the null character

novusnota committed:
test: unicode code point escapes on control codes

Didn't add tests for dynamic generation of strings with unicode escapes
higher than 0x7F due to some weird unability to match the strings from
Tact over to ones in Jest/TypeScript. Note, that dumping said strings
and passing them around everywhere in Tact works just fine. It's
evaluation of said code points in strings that works differently in
TypeScript and TVM (I suspect different encoding standards apply).
